subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -56,6 +56,15 @@
 
 #define DISCOVER_PREFACE	"</.well-known/core>;ct=40"
 
+#define RESOURCE_TYPE		";rt=\"oma.lwm2m\""
```

Comment:
I don't have any specific reason to implement alternative path.
I was intended to report that client supports JSON format.
To do that, the tag ";ct=11543" has to be appended on "</>".
However, one LwM2M server implementation (wakaama) that I used for testing is picky on "</>;ct=11543" and will reject client registration.
That's why I add "rt=\"oma.lwm2m\"" to the REG_PREFACE.

I suppose that we can set REG_PREFACE = "" when JSON format is not supported. For we are not going to indicate any alternative path to the server.